script/testing/oltpbench/README.md

```diff
@@ -32,34 +108,59 @@ Let the base directory be `noisepage/script/testing/oltpbench`
   - End-to-end performance
   - Nightly
 
-  For more details, please refer to [Configuration file for OLTPBench](https://github.com/cmu-db/noisepage/blob/master/script/testing/oltpbench/configs/README.md)
+  For more details, check the [config files](https://github.com/cmu-db/noisepage/blob/master/script/testing/oltpbench/README.md#config-files) section below for more details
 - `run_oltpbench.py`: entry point to run the OLTP bench tests
 - `test_oltpbench.py`: defines the `TestOLTPBench` class which manage the lifecycle of a test suite
 - `test_case_oltp.py`: defines the `TestOLTPBench` class which manage the lifecycle of a test case
 - `utils.py`: defines a list of utility functions specifically used by OLTP bench tests
 
-## Test workflow
-- Start the NoisePage DB process as a Python subprocess
-  - Try to locate the NoisePage binary
-  - Kill all the lingering processes on the NoisePage port
-  - Start the NoisePage DB process
-- Read the test case configs
-- `run_pre_suite`: Pre test suite tasks
-  - Clean the possible residual local [oltpbench](https://github.com/oltpbenchmark/oltpbench) workspace
-  - Git clone the [oltpbench](https://github.com/oltpbenchmark/oltpbench) repo
-  - [optional] *Checkout to the specified branch*
-- Iterate through all the test cases
-  - `run_pre_test`: Pre test case tasks
-    - Create the database and tables for the OLTP benchmark specified
-    - Load the data to tables
-  - Run the test case command as a subprocess
-    - Collect the memory info by using the [PeriodicTask](https://github.com/cmu-db/noisepage/blob/master/script/testing/util/periodic_task.py) in a separate thread
-      - Collect `RSS` and `VMS` by default
-      - Collect every `5` seconds by default
-      - The memory info is stored in a Python dictionary in memory in runtime
-  - `run_post_test`: Post test case tasks
-    - If it is part of the Jenkins nightly build, the result results should be stored
-        - Parse the testing results files by [oltpbench](https://github.com/oltpbenchmark/oltpbench) and format them in JSON
-        - Add the memory info to `incremental_metrics` and compute the average metrics to add to the `metrics` in JSON payload
-        - Send a POST request to the Django API
-- `run_post_suite`: Post test case suite
\ No newline at end of file
+## Config files
+To run a OLTPBench test, you should run the `run_oltpbench.py --config-file=config.json`. 
+
+### Fields
+In the configuration file, those information are required:
+- A list of test cases in `testcases`
+- The benchmarks and options in each test case, required by the oltpbench's workload descriptor file
+  - The `run_oltpbench` script will run all test cases in the configuration file sequentially. 
+  - The `loop` key in the configuration file is used to duplicate the test case with different options.
+  - The `server_args` filed in the configuration specify the server command line args.
+
+### Example:
+
```

Comment:
A short description of what test cases this config will run would probably be really helpful.

Reply:
i am not sure which part is confusing here. because i think the configuration file itself is quite self-explaining and a description is basically repeating what the configuration file says.

Reply:
i added a few lines here anyway, not sure if it what you want.